apps/job-core/src/amqp_to_redis.js

```diff
@@ -0,0 +1,76 @@
+import redis from 'redis'
+import { v4 as uuid } from 'uuid'
+import { promisify } from 'util'
+
+const sleep = promisify(setTimeout)
+
+class AMQPToRedis {
+  constructor(redis_url) {
+    this.client = redis.createClient({ url: redis_url })
+    this.lpop = promisify(this.client.lpop).bind(this.client)
+    this.quit = promisify(this.client.quit).bind(this.client)
+    this.closed = false
+    this.ack_callbacks = {}
+    this.exec_message = promisify((content, message_callback, callback) => {
+      const id = uuid()
+      this.ack_callbacks[id] = callback
+      message_callback({id: id, content: content})
+    }).bind(this)
+  }
+
+  queueName(queue) {
+    return 'queue_' + queue
+  }
+
+  ack(message) {
+    const callback = this.ack_callbacks[message.id]
+    if (callback) {
+      this.ack_callbacks[message.id] = null
+      callback()
+    }
+  }
+
+  nack(message, _allUpTo = false, _requeue = false) {
```

Comment:
Are `_allUpTo` or `_requeue` used externally?

Reply:
AMQP API :)

Reply:
Generally parameters starting with `_` are private, so I'm not quite sure they're used as part of the AMQP API.  Moreover, in javascript, you don't need to declare this parameters in order to permit other code to pass them.  Since you're not used, you can just delete them with no side-effects.

```
 0 ✓ isabelle.cowan-bergman@mbp-de-isabelle.home oauth2_proxy $ node
Welcome to Node.js v12.13.0.
Type ".help" for more information.
> function foo(blah) {
... console.log(blah)
... }
undefined
> foo("blah")
blah
undefined
> foo("blah", "extra")
blah
undefined
> foo("blah", extra => "extra")
blah
undefined
```